backend/src/api/ApiHandler.py

```diff
@@ -0,0 +1,60 @@
+from typing import Tuple
+
+from backend.src.MapAPI import MapAPI
+
+from backend.src.api.nominatim import get_lat_long
+from backend.src.api.overpass import nearby_point
+from backend.src.api.osrm import determine_travel_time
+from backend.src.api.osrm import TravelOptions
+
+
+class ApiHandler(MapAPI):
+    """
+    Implements MapAPI by connecting to three APIs: nominatim, overpass, and OSRM
+    Nominatim handles converting addresses to lat, long pairs
+    Overpass handles searching for locations in a radius around a center
+    OSRM handles routing and time estimations
+    All are free and open source
+    """
+
+    def convert(self, loc):
+        """
+        Helper method that converts a given address to a latitude, longitude representation.
+        If the given location is already lat, long nothing happens
+        :param loc: The given location, either address or lat, long form
+        :return: None if error, else the lat, long tuple
+        """
+        if not isinstance(loc, Tuple):
+            # Assume it's an address and convert
+            loc = get_lat_long(loc)
+        return loc
+
+    def get_travel_time(self, loc_1, loc_2) -> float:
+        loc_1 = self.convert(loc_1)
+        if loc_1 is None:
+            return None
+
+        loc_2 = self.convert(loc_2)
+        if loc_2 is None:
+            return None
+
+        travel_time = determine_travel_time(loc_1, loc_2, TravelOptions.WALK)
```

Comment:
Is walking going to be our default option? (Obviously it would be a stretch goal to allow multiple options here, but wouldn't it be the most versatile to default to driving? If we agreed to walking I honestly don't recall that)

Reply:
I think we said walking, but we could do driving. The only gross thing is that OSRM has different endpoints for each options so I'd have to go and make some changes to allow for both lol

Reply:
Let's leave it as is then.